github-runner-provisioner/internal/aws/ec2.go

```diff
@@ -36,7 +36,7 @@ func (c *Ec2Client) GetInstances(filter []types.Filter) ([]*InstanceDetails, err
 			for _, instance := range reservation.Instances {
 				label, err := getActionRunnerLabel(instance)
 				if err != nil {
-					log.Printf("Error getting runner tag for instance %s: %v\n", *instance.InstanceId, err)
+					log.Errorf("Error getting runner tag for instance %s: %v\n", *instance.InstanceId, err)
```

Comment:
This works for now, but the advantage of having structured logs is that you can define fields in the logs, which make search/filtering easier in GCP
For example, instead of logging
```
{
  "severity": "ERROR",
  "message": "Error getting runner tag for instance SOME_INSTANCE: ERROR MESSAGE"
}
```

You could have
```
{
  "severity": "ERROR",
  "message": "Error getting runner tag: ERROR MESSAGE",
  "instanceId": "SOME_INSTANCE"
}
```

I think that putting custom fields under `logging.googleapis.com/labels` is better (query performance) but I want to try it out before going there:
```
{
  "severity": "ERROR",
  "message": "Error getting runner tag: ERROR MESSAGE",
  "logging.googleapis.com/labels": {
    "instanceId": "SOME_INSTANCE"
  }
}
```

Reply:
Yeah, labels seem like they would be very useful here. I think that's a bit out of scope for this effort, but I'd love to play with it during cooldown. Getting all of our infra into a consistently better logging structure would be *excellent* as a cooldown project.